### PR TITLE
TELL A DEVELOPER WHEN SET AWAY: As a user, I want to have a message from the app when somebody says I am away, so that I can set myself as available when I am back

### PR DESCRIPTION
**Pull request by Reinkaos**
**Acceptance Criteria**
- When somebody is set as away, the person is receiving a slack message telling him
- A cronjob must send a reminder every 7 days
- Optional: Add a button in the message to set the developer status as available

### CODE REVIEW

File: app/services/slack/action/set_developer.rb [hunk 1]
```diff
@@ -21,6 +21,13 @@ def initialize(slack_workspace, developer, attributes)
         else
           developer.update(away: away_status == 'away')
           @text = "Developer \"#{developer_tag}\" is now #{away_status}"
+
+          Slack::Api::ChatPostMessage.new(
+            channel: developer['slack_id'],
+            text: "Your status on the Code Review app has been set to #{away_status}.",
```
Comment (paul-mesnilgrente) [inline app/services/slack/action/set_developer.rb:27]: You could add the command to set the dev as available when that's done :)

File: app/models/developer.rb [hunk 2]
```diff
@@ -37,4 +39,9 @@ def self.queue
       .where(away: false)
       .order(max_updated_at: :asc)
   end
+
+  def status
+    return 'away' if away
+    'available'
```
Comment (paul-mesnilgrente) [inline app/models/developer.rb:45]: I think I'd prefer the syntax `away ? 'away' : 'available'` but I guess it's a matter of taste haha

File: app/services/slack/argument_parser.rb [hunk 3]
```diff
@@ -48,6 +49,7 @@ class RestOptions
         developer_get: %i[persisted_developer],
         developer_list: %i[],
         developer_delete: %i[persisted_developer],
+        developer_set: %i[],
```
Comment (paul-mesnilgrente) [inline app/services/slack/argument_parser.rb:52]: I think you can add `persisted_developer` developer again :)

File: app/services/slack/argument_parser.rb [hunk 4]
```diff
@@ -196,10 +200,12 @@ def define_options(parser)
...
-        parser.separator '        Add:    /cr -a add    -o developer -d @dave'
-        parser.separator '        List:   /cr -a list   -o developer'
-        parser.separator '        Get:    /cr -a get    -o developer -d @dave'
-        parser.separator '        Remove: /cr -a delete -o developer -d @dave'
+        parser.separator '        Add:           /cr -a add    -o developer -d @dave'
+        parser.separator '        List:          /cr -a list   -o developer'
+        parser.separator '        Get:           /cr -a get    -o developer -d @dave'
+        parser.separator '        Remove:        /cr -a delete -o developer -d @dave'
+        parser.separator '        Set away:      /cr -a set    -o developer -d @dave --away' 
+        parser.separator '        Set available: /cr -a set    -o developer -d @dave --available'
```
Comment (paul-mesnilgrente) [inline app/services/slack/argument_parser.rb:208]: Nice ;)

File: app/services/slack/argument_parser.rb (hunk 3 above)
Comment (Reinkaos) [reply app/services/slack/argument_parser.rb:52]: nice sure thing :D

File: app/models/developer.rb (hunk 2 above)
Comment (Reinkaos) [reply app/models/developer.rb:45]: I'm not a ternary fan haha

**Pull request merged**